app/lib/import_export.rb

```diff
@@ -11,497 +12,226 @@ module ImportExport # rubocop:todo Metrics/ModuleLength
-                                                           jurisdiction_path jurisdiction_name assigned_user]),
-              rct_node(:patients, 'Monitoring Period', %i[last_date_of_exposure continuous_exposure symptom_onset symptom_onset_defined_by
-                                                          extended_isolation end_of_monitoring closed_at monitoring_reason expected_purge_ts]),
-              rct_node(:patients, 'Reporting Info', %i[responder_id head_of_household pause_notifications last_assessment_reminder_sent])
-            ]
-          }
-        ]
-      }
-    ]
-  }.freeze
-
-  ASSESSMENTS_EXPORT_OPTIONS = {
-    label: 'Reports',
-    nodes: [rct_node(:assessments, 'Reports', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id symptomatic
-                                                 who_reported created_at updated_at symptoms])]
-  }.freeze
-
-  LABORATORIES_EXPORT_OPTIONS = {
-    label: 'Lab Results',
-    nodes: [rct_node(:laboratories, 'Lab Results', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id lab_type
-                                                      specimen_collection report result created_at updated_at])]
-  }.freeze
-
-  CLOSE_CONTACTS_EXPORT_OPTIONS = {
-    label: 'Close Contacts',
-    nodes: [rct_node(:close_contacts, 'Close Contacts', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id first_name
-                                                           last_name primary_telephone email contact_attempts last_date_of_exposure assigned_user notes
-                                                           enrolled_id created_at updated_at])]
-  }.freeze
-
-  TRANSFERS_EXPORT_OPTIONS = {
-    label: 'Transfers',
-    nodes: [rct_node(:transfers, 'Transfers', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id who from_jurisdiction
-                                                 to_jurisdiction created_at updated_at])]
-  }.freeze
-
-  HISTORIES_EXPORT_OPTIONS = {
-    label: 'History',
-    nodes: [rct_node(:histories, 'History', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id created_by history_type
-                                               comment created_at updated_at])]
-  }.freeze
-
-  CUSTOM_EXPORT_OPTIONS = {
-    patients: PATIENTS_EXPORT_OPTIONS,
-    assessments: ASSESSMENTS_EXPORT_OPTIONS,
-    laboratories: LABORATORIES_EXPORT_OPTIONS,
-    close_contacts: CLOSE_CONTACTS_EXPORT_OPTIONS,
-    transfers: TRANSFERS_EXPORT_OPTIONS,
-    histories: HISTORIES_EXPORT_OPTIONS
-  }.freeze
+  # Gets data for this batch of patients that may not have already been present in the export config (such as specific symptoms).
+  def get_field_data(config)
+    data = config[:data].deep_dup
```

Comment:
Would it be better to check first to see if any update is required before incurring the `deep_dup` cost?

Reply:
hmm some of these checks are a little more complicated (for being a one line check or something before calling `deep_dup`) such as the one for `update_headers` where 2 checks are performed for each data type in a loop, would it be possible to only `deep_dup` part of the `data` hash?